backend/app/api_views.py

```diff
@@ -181,3 +187,74 @@ def campaign_finance_donor_subset(request, donor_name):
     campaign_finances = CampaignFinance.objects.filter(donor_name=donor_name)
     serializer = CampaignFinanceSerializer(campaign_finances, many=True)
     return Response(serializer.data)
+
+
+@api_view(['GET'])
+def get_lokniti_codebook(request):
+    """
+    API endpoint to get codebook
+    """
+    responses = LoknitiCodebook.objects.all()
+    serializer = LoknitiCodebookSerializer(responses, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['GET'])
+def get_lokniti_responders(request):
+    """
+    API endpoint to get responders
+    """
+    responses = LoknitiResponders.objects.filter(PC_id=1, PS_id=1)
+    serializer = LoknitiRespondersSerializer(responses, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['GET'])
+def get_responders_by_constituency(request, election_year, state_name, pc_id):
+    """
+    API endpoint to get responders by constituency
+    """
+    state_name = state_name.replace("_", " ")
+    if election_year == 2009:
+        state_name = state_name.upper()
+    responses = LoknitiResponders.objects.filter(
+        election_year=election_year, state_name=state_name, PC_id=pc_id)
```

Comment:
I think the issue with the map is caused by `state_name=state_name` in the filter. I'm not sure how you were able to get any information to appear on the map (when I first loaded the map, everything was gray) since right now, the `state_name` that is saved in the database for a Responder includes the state id (ex. "34: Chhattisgarh" instead of "Chhattisgarh") so checking `state_name=state_name` should not match any responder. The map looks like this after changing `state_name=state_name` to `state_name__icontains=state_name`:

<img width="1175" alt="image" src="https://github.com/dhmit/data-driven-democracy/assets/55002917/a3252e0c-d643-4e0e-9c99-d49d1406e545">

Reply:
Oh that is interesting. In my load_responders.py, I think I had it remove the number and only saved the state_name in the table (which now looking back, probably wasn't the best idea).
![image](https://github.com/dhmit/data-driven-democracy/assets/152655014/7fe99d88-40bb-413f-8fc2-344030c3f2e6)

Reply:
Ah, sorry, I missed that when I was refactoring. Ultimately, it's probably better to use something like an [enumeration type](https://docs.djangoproject.com/en/5.0/ref/models/fields/#enumeration-types) to keep the state name and ID easily accessible from the Responder. Do you think the map looks okay now? If so, I'll just leave the state name and ID together for now.

Reply:
It seems like 2014 and 2019 is missing a lot of data, but I am not sure if that's an issue with the code or it's just all the data that is in the csv's. 

I'll quickly fix an issue with the data being displayed on the side and put all the color data on the backend.